src/codec/json.rs

```diff
@@ -12,35 +12,60 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::cmp::max;
+use std::{cmp::max, marker::PhantomData};
 
 use super::prelude::*;
+use tremor_script::utils::sorted_serialize;
 use tremor_value::AlignedBuf;
 
-pub struct Json {
+/// Sorting for JSON
+pub trait Sorting: Sync + Send + Copy + Clone + 'static {
+    const SORTED: bool;
```

Comment:
Disgusting, I love it, very nice indeed! 🚀

Reply:
:D ja and efficient (see https://rust.godbolt.org/z/3nTsd1s1K)